core/src/env_platform/db_handler.bal

```diff
@@ -73,10 +50,44 @@ function saveApplication(TreeRemovalForm form) returns error? {
                     "district": form.district,
                     "nameOfTheLand": form.nameOfTheLand,
                     "planNumber": form.planNumber,
-                    "area": locations,
-                    "treeInformation": treeInformation
+                    "area": getAreaJsonArray(form.area),
+                    "treeInformation": getTreeInformationJsonArray(form.treeInformation)
                 }
             ]
     };
     return applicationCollection->insert(application);
 }
+
+# The `deleteApplication` function will delete application drafts only which status is save.
+# 
+# + applicationId - The Id of the application which must be deleted
+# + return - This function will return null if application is deleted from the database or else return mongodb:DatabaseError
+# array index out of bound if there are no application with the specific application Id.
```

Comment:
The return should either be true, false or an error we need to check the delete variable and return true or false.